packages/components/src/segmented-control/component.tsx

```diff
@@ -49,6 +58,7 @@ function SegmentedControl(
 	const containerRef = useRef();
 	const [ resizeListener, sizes ] = useResizeAware();
 
+	const id = useUniqueId( idProp );
 	const radio = useRadioState( {
 		baseId: baseId || id,
```

Comment:
According to the [docs](https://developer.wordpress.org/block-editor/reference-guides/components/base-control/#id), the `id` of `BaseControl` should match "the `id` of the element to which labels and help text are being generated".

It seems like this wouldn't be the case anymore if `baseId` is different from `id`.

What is the reason to have both a `baseId` and an `id` prop in this component?

Reply:
I removed the `id` as it doesn't make sense to have one in our case - related comment above: https://github.com/WordPress/gutenberg/pull/34017#discussion_r687623770.

>What is the reason to have both a baseId and an id prop in this component?

That I'm not aware of.. @diegohaz might have some info for that? 🤔 . This code is from the `g2` and thought it had to do with internals from `Reakit`..